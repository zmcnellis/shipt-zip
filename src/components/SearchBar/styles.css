/* Search --------------------------------------------------------------------*/
.search-wrapper {
  display: flex;
  flex-wrap: wrap;
  flex-direction: column;
  position: relative;
  margin: 0 auto;
  margin-top: 25px;
  margin-bottom: 25px;
  width: 80%;
  max-width: 300px;
}

.search-bar {
  /* order: 1; */
  color: #888;
  background-color: #fff;
  width: 220px;
  height: 48px;
  border-radius: 2px;
  width: 100%;
}

.search-bar:focus-within {
  color: #333;
}

.search-icon {
  line-height: 48px;
  width: 10%;
  box-sizing: border-box;
  text-align: center;
  position: relative;
}

.search-icon-left {
  float: left;
}

.search-icon-right {
  float: right;
}

.search-field {
  display: inline-block;
  width: 80%;
  height: 100%;
  margin: 2px 0;
  line-height: 45px;
}

.search-field-input {
  color: #888;
  border: 0;
  font-size: 14px;
  outline: none;
  width: 100%;
  font-size: 16px;
  font-weight: bold;
  margin: 0;
  padding: 0;
}

.search-field-input:focus {
  color: #333;
}

.search-field-input.warning:focus {
  color: #ef7c42;
}

.search-button {
  /* order: 3; */
  width: 100%;
  margin-top: 14px;
}

.error-text {
  /* order: 2; */
  width: 100%;
  color: #ef7c42;
  padding-top: 5px;
  margin-bottom: 0;
  font-weight: bold;
  text-transform: uppercase;
  font-size: 11px;
}

/* Media queries -------------------------------------------------------------*/
@media (min-width: 700px) {
  /* search */
  .search-wrapper {
    flex-direction: row;
    width: 40%;
    min-width: 340px;
  }

  .search-button {
    width: 32%;
    margin-top: 0;
    border-radius: 0 2px 2px 0;
    /* order: 2; */
  }

  .search-bar {
    width: 68%;
    border-radius: 2px 0 0 2px;
  }
}
