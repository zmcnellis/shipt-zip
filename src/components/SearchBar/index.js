import React, { Component } from 'react';
import PropTypes from 'prop-types';
import './styles.css';

class SearchBar extends Component {

  // store input value in component's state
  constructor(props) {
    super(props);
    this.state = {
      inputValue: ''
    };
    this.clearValidationErrors = false; // when typing in the search box, all validation errors should disappear
    this.updateValue = this.updateValue.bind(this);
  }

  // submit search term to parent component for processing
  handleSubmit() {
    this.props.handleSubmit(this.state.inputValue);
    this.clearValidationErrors = false;
  }

  // after submitting an invalid zip, any changes to the field should cause the validation errors to disappear
  updateValue(evt) {
    this.setState({
      inputValue: evt.target.value
    });
    if (this.props.hasError)
      this.clearValidationErrors = true;
  }

  // display search bar depending on validation error state
  displaySearch() {
    if (this.props.hasError && !this.clearValidationErrors) {
      return (
        <div className="search-wrapper">
          <div className="search-bar">
            <div className="search-icon warning search-icon-left">
              <i className="fas fa-map-marker-alt"></i>
            </div>
            <div className="search-field">
              <input ref={(input) => { this.props.setSearchReference(input); }} className="search-field-input warning" name="search" type="text" autoComplete="off" maxLength="5" placeholder="Enter ZIP Code" onChange={evt => this.updateValue(evt)}/>
            </div>
            <div className="search-icon warning search-icon-right">
              <i className="fas fa-exclamation-circle"></i>
            </div>
          </div>
          <a className="button button-primary search-button" onClick={this.handleSubmit.bind(this)}>Get Started</a>
          <p className="error-text">This ZipCode is invalid! Please try again.</p>
        </div>
      );
    }
    else {
      return (
        <div className="search-wrapper">
          <div className="search-bar">
            <div className="search-icon search-icon-left">
              <i className="fas fa-map-marker-alt"></i>
            </div>
            <div className="search-field">
              <input ref={(input) => { this.props.setSearchReference(input); }} className="search-field-input" name="search" type="text" autoComplete="off" maxLength="5" placeholder="Enter ZIP Code" onChange={evt => this.updateValue(evt)}/>
            </div>
            <div className="search-icon warning search-icon-right" style={{visibility: 'hidden'}}>
              <i className="fas fa-exclamation-circle"></i>
            </div>
          </div>
          <a className="button button-primary search-button" onClick={this.handleSubmit.bind(this)}>Get Started</a>
        </div>
      );
    }
  }

  // render
  render() {
    return this.displaySearch();
  }
}

SearchBar.propTypes = {
  handleSubmit: PropTypes.func.isRequired,
  hasError: PropTypes.bool.isRequired,
  setSearchReference: PropTypes.func.isRequired
};

export default SearchBar;
