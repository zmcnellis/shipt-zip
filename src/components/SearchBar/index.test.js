import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import SearchBar from '.';

describe('SearchBar', () => {
  it('should render correctly', () => {
    const output = shallow(
      <SearchBar hasError={false} handleSubmit={() => "mock"} setSearchReference={() => "mock"}/>
    );
    expect(output).toMatchSnapshot();
  });

  it('should have an input field', () => {
    const output = shallow(
      <SearchBar hasError={false} handleSubmit={() => "mock"} setSearchReference={() => "mock"}/>
    );
    expect(output.find('input[type="text"]').exists()).toBeTruthy();
  });

  it('should have a search button', () => {
    const output = shallow(
      <SearchBar hasError={false} handleSubmit={() => "mock"} setSearchReference={() => "mock"}/>
    );
    expect(output.find('.search-button').exists()).toBeTruthy();
  });

  it('should toggle the warning class on error', () => {
    const output = shallow(
      <SearchBar hasError={true} handleSubmit={() => "mock"} setSearchReference={() => "mock"}/>
    );
    expect(output.find('.search-icon').every('.warning')).toBeTruthy();
    expect(output.find('.search-field-input').hasClass('warning')).toBeTruthy();
  });
});
