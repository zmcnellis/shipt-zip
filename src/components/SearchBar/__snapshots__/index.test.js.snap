// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SearchBar
    handleSubmit={[Function]}
    hasError={false}
    setSearchReference={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="search-bar"
>
          <div
                    className="search-icon search-icon-left"
          >
                    <i
                              className="fas fa-map-marker-alt"
                    />
          </div>
          <div
                    className="search-field"
          >
                    <input
                              autoComplete="off"
                              className="search-field-input"
                              maxLength="5"
                              name="search"
                              onChange={[Function]}
                              placeholder="Enter ZIP Code"
                              type="text"
                    />
          </div>
          <div
                    className="search-icon warning search-icon-right"
                    style={
                              Object {
                                        "visibility": "hidden",
                                      }
                    }
          >
                    <i
                              className="fas fa-exclamation-circle"
                    />
          </div>
</div>,
        <a
          className="button button-primary search-button"
          onClick={[Function]}
>
          Get Started
</a>,
      ],
      "className": "search-wrapper",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="search-icon search-icon-left"
>
              <i
                            className="fas fa-map-marker-alt"
              />
</div>,
            <div
              className="search-field"
>
              <input
                            autoComplete="off"
                            className="search-field-input"
                            maxLength="5"
                            name="search"
                            onChange={[Function]}
                            placeholder="Enter ZIP Code"
                            type="text"
              />
</div>,
            <div
              className="search-icon warning search-icon-right"
              style={
                            Object {
                                          "visibility": "hidden",
                                        }
              }
>
              <i
                            className="fas fa-exclamation-circle"
              />
</div>,
          ],
          "className": "search-bar",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <i
                className="fas fa-map-marker-alt"
/>,
              "className": "search-icon search-icon-left",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "fas fa-map-marker-alt",
              },
              "ref": null,
              "rendered": null,
              "type": "i",
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <input
                autoComplete="off"
                className="search-field-input"
                maxLength="5"
                name="search"
                onChange={[Function]}
                placeholder="Enter ZIP Code"
                type="text"
/>,
              "className": "search-field",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "autoComplete": "off",
                "className": "search-field-input",
                "maxLength": "5",
                "name": "search",
                "onChange": [Function],
                "placeholder": "Enter ZIP Code",
                "type": "text",
              },
              "ref": [Function],
              "rendered": null,
              "type": "input",
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <i
                className="fas fa-exclamation-circle"
/>,
              "className": "search-icon warning search-icon-right",
              "style": Object {
                "visibility": "hidden",
              },
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "fas fa-exclamation-circle",
              },
              "ref": null,
              "rendered": null,
              "type": "i",
            },
            "type": "div",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Get Started",
          "className": "button button-primary search-button",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Get Started",
        "type": "a",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="search-bar"
>
            <div
                        className="search-icon search-icon-left"
            >
                        <i
                                    className="fas fa-map-marker-alt"
                        />
            </div>
            <div
                        className="search-field"
            >
                        <input
                                    autoComplete="off"
                                    className="search-field-input"
                                    maxLength="5"
                                    name="search"
                                    onChange={[Function]}
                                    placeholder="Enter ZIP Code"
                                    type="text"
                        />
            </div>
            <div
                        className="search-icon warning search-icon-right"
                        style={
                                    Object {
                                                "visibility": "hidden",
                                              }
                        }
            >
                        <i
                                    className="fas fa-exclamation-circle"
                        />
            </div>
</div>,
          <a
            className="button button-primary search-button"
            onClick={[Function]}
>
            Get Started
</a>,
        ],
        "className": "search-wrapper",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="search-icon search-icon-left"
>
                <i
                                className="fas fa-map-marker-alt"
                />
</div>,
              <div
                className="search-field"
>
                <input
                                autoComplete="off"
                                className="search-field-input"
                                maxLength="5"
                                name="search"
                                onChange={[Function]}
                                placeholder="Enter ZIP Code"
                                type="text"
                />
</div>,
              <div
                className="search-icon warning search-icon-right"
                style={
                                Object {
                                                "visibility": "hidden",
                                              }
                }
>
                <i
                                className="fas fa-exclamation-circle"
                />
</div>,
            ],
            "className": "search-bar",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <i
                  className="fas fa-map-marker-alt"
/>,
                "className": "search-icon search-icon-left",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "fas fa-map-marker-alt",
                },
                "ref": null,
                "rendered": null,
                "type": "i",
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <input
                  autoComplete="off"
                  className="search-field-input"
                  maxLength="5"
                  name="search"
                  onChange={[Function]}
                  placeholder="Enter ZIP Code"
                  type="text"
/>,
                "className": "search-field",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "autoComplete": "off",
                  "className": "search-field-input",
                  "maxLength": "5",
                  "name": "search",
                  "onChange": [Function],
                  "placeholder": "Enter ZIP Code",
                  "type": "text",
                },
                "ref": [Function],
                "rendered": null,
                "type": "input",
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <i
                  className="fas fa-exclamation-circle"
/>,
                "className": "search-icon warning search-icon-right",
                "style": Object {
                  "visibility": "hidden",
                },
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "fas fa-exclamation-circle",
                },
                "ref": null,
                "rendered": null,
                "type": "i",
              },
              "type": "div",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Get Started",
            "className": "button button-primary search-button",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Get Started",
          "type": "a",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
